graph.rb

```diff
@@ -45,3 +57,6 @@ def to(edge)
     edge.last
   end
 end
+
+class UnsortableCyclicGraphError < StandardError
```

Comment:
Since this error is used by the `Graph` class, I'd probably put it at the top of the file.

You can also achieve the same effect with a one-liner:
```ruby
UnsortableCyclicGraphError = Class.new(StandardError)
```